docs/txp.md

```diff
@@ -0,0 +1,486 @@
+* Signature `sig` is a valid signature of transaction (validity
+  checked with `pk`) As in public key witness, signature inside redeem
+  witness must be given for `TxSigData` (which is basically the same
+  as `Hash Tx`). `SignRedeemTx` tag should be. The sequence of bytes
+  to sign is: `02 <magic> HASH_CBOR` (it differs
+  from [Public key witness](#public-key-witness) in only one byte).
+
+
+###### Unknown witness
+
+* If `verifyAllIsKnown` is `True`, witness is invalid
+* If witness has unknown type with tag `t`, corresponding address must
+  have unknown type with tag `t`
+
+##### Sums check
+
+This check is omitted if (and only if):
+* There is an unknown transaction input
+
+For each input and output we know its value (i.e. associated balance). We compute sum of all inputs and all outputs.
+
+* If sum of outputs is greater than sum of inputs, tx is considered invalid
+* Otherwise, check succeeds, difference between sum of inputs and sum
+  of outputs is considered to be *transaction fee*.
+
+##### Transaction fee check
+
+This check is omitted if (and only if) one of conditions met:
+* There is an unknown transaction input
+* All inputs correspond to `Redeem` addresses
+  - Implication of this rule is that one can redeem ADA without paying fee
+* Currently adopted tx fee policy is unknown
+
+For this check we know:
+
+* Transaction fee `txFee` (from *Sums check*)
+* Transaction size `txSize` (from *Transaction size check*)
+* Currently adopted tx fee policy (as part of `BlockVersionData`), which is of known type
+
+###### Size-linear policy
+Only policy supported by Cardano SL v1.0 is `TxFeePolicyTxSizeLinear`.
+
+For it, we need to compute minimal required fee for this transaction.
+This value is computed using fixed-precision arithmetic (precision is 1e-9) using formula:
+
+```
+minFee = a + b * txSize
+```
+
+There are two cases check fails (and transaction rejected):
+
+* Minimal fee is negative or is greater than maximal number of coins in the system
+
+   ```minFee < 0 || minFee > maxBound @Coin```
+   - It probably indicates there is a mistake in currently adopted fee
+   policy
+* Actual fee is less than the minimal fee
+
+   ```txFee < minFee```
+
+### GState modification
+
+If all checks above pass, we modify GState appropriately. We modify
+stakes and UTXO.
+
+#### UTXO modification
+
+UTXO modification is trivial:
+
+* All `TxInUtxo` inputs are deleted from UTXO
+* For each output `out` with index `i` we put `(TxInUtxo (hash tx) i,
+  out)` into UTXO.
+
+Note, that `TxInUtxo` is only input type supported in CSL v1.0. For unknown inputs GState is not modified.
+
+#### Computing stakes for transaction output
+
+Each `txOut` can be converted to a list of `(StakeholderId, Coin)` pairs.
+
+This is being done with considering `AddrStakeDistribution` attribute of address corresponding to `txOut`:
+
+- For `BootstrapEraDistr` the value `val` of `txOut` is distributed among
```

Comment:
IMHO this section is described very informally, what are weights? What is `bootDustThreshold` and why is it needed? What does mean `The first stakeholder may receive slightly more than others`?

Reply:
Not principal, you can rewrite if you want

Reply:
lol, mne len'